net/portmapper/igd_test.go

```diff
@@ -21,10 +24,13 @@ type TestIGD struct {
 	upnpConn net.PacketConn // for UPnP discovery
 	pxpConn  net.PacketConn // for NAT-PMP and/or PCP
 	ts       *httptest.Server
+	t        *testing.T
 
-	doPMP  bool
-	doPCP  bool
-	doUPnP bool // TODO: more options for 3 flavors of UPnP services
+	// do* will log which packets are sent, but will not reply to unexpected packets.
+
+	doPMP  syncs.AtomicBool
```

Comment:
why do these need to be atomic? they're written before they're ever read, no?

Reply:
I don't think they need to be, but the linux-race test reports it as a race, so I just changed it to atomic